Clarify DNS server address handling, add references (#133)
tfpauly/draft-happy-eyeballs-v3: draft-ietf-happy-happyeyeballs-v3.md

If multiple DNS server addresses are configured for the current
network, the client may have the option of sending its DNS queries
over IPv4 or IPv6. This is discussed in

> {{Section 4.3 of ?I-D.ietf-dnsop-3901bis}}, which points to

Reviewer: nit: This feels like an unnecessary reference if we're saying that it just references the other document.

Destination Address Selection as defined in {{!RFC6724}}.

> In keeping with the Happy Eyeballs approach, an implementation

Reviewer: ```suggestion
In accordance with the Happy Eyeballs approach, an implementation
```

> In accordance with the Happy Eyeballs approach, an implementation

Reviewer 2: The use if "In accordance with" sounds weird to me. But taking a step back, what are we trying to accomplish here? Communication to the DNS resolver feels out of scope for this document

Author: The selection of address family used to talk to the resolver was part of HEv2 as well (I thought you wrote that part, David!), so this is mainly trying to update and rephrase that.

Reviewer 2: How dare you suggest that past me was right!? That guy keeps making mistakes that I have to deal with

of a stub resolver that supports Happy Eyeballs SHOULD additionally
prefer using IPv6 resolver addresses when possible. This can be
accomplished by sorting IPv6 addresses first prior to running
the Destination Address Selection sorting from {{!RFC6724, Section 6}}.
Note that this preference for IPv6 is not referring to the sending of
AAAA or A queries, but rather the address of the DNS server itself

> and IP version used to transport DNS messages.

Reviewer: nit:
```suggestion
and the IP version used to transport DNS messages.
```

> Also in keeping with Happy Eyeballs, when DNS queries are sent

Reviewer: ```suggestion
Also in accordance with Happy Eyeballs, when DNS queries are sent to
```
